components/builder-api/src/server/handlers.rs

```diff
@@ -119,6 +125,60 @@ pub fn job_group_promote(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+pub fn validate_docker_credentials(req: &mut Request) -> IronResult<Response> {
+    let json_body = req.get::<bodyparser::Json>();
+
+    let body = match json_body {
+        Ok(Some(b)) => b,
+        Ok(None) => {
+            debug!("Error: Missing request body");
+            return Ok(Response::with(status::BadRequest));
+        }
+        Err(err) => {
+            debug!("Error: {:?}", err);
+            return Ok(Response::with(status::BadRequest));
+        }
+    };
+
+    if !body["username"].is_string() || !body["password"].is_string() {
+        debug!("Error: Missing username or password");
+        return Ok(Response::with(status::BadRequest));
+    }
+
+    // There's probably a better way to do this. Suggestions?
```

Comment:
The only other thing that occurs to me is shelling out to `docker` and running e.g. `docker login` and checking its exit code.  One nice thing about that is if they change their API in the future, we don't have to change our API call.  Honestly though, it's probably not a big deal.  This looks fine too.

Reply:
I actually did look into that! But it wasn't clear `docker` would be installed on the API nodes, and it didn't seem necessary to install it, etc., just for this use case.

Reply:
Yeah, good points